src/user_lists/serializers.py

```diff
@@ -15,6 +22,137 @@ class Meta:
             "updated_date",
             "created_by",
             "updated_by",
+            "items_count",
         ]
         read_only_fields = ["id", "created_date", "updated_date", "created_by", "updated_by"]
 
+    def get_items_count(self, obj):
+        if hasattr(obj, '_prefetched_objects_cache') and 'items' in obj._prefetched_objects_cache:
+            return len([item for item in obj._prefetched_objects_cache['items'] if not item.is_removed])
+        return obj.items.filter(is_removed=False).count()
+
+
+class ListItemSerializer(DefaultAuthenticatedSerializer):
+    unified_document = serializers.SerializerMethodField()
+
+    class Meta:
+        model = ListItem
+        fields = ["id", "parent_list", "unified_document", "created_date", "created_by"]
+        read_only_fields = ["id", "created_date", "created_by"]
+        extra_kwargs = {
+            'unified_document': {'required': True}
+        }
+
+    def get_fields(self):
```

Comment:
We don't have this pattern anywhere in the codebase so I request we don't do something like this.

Serializers should be "dumb" we should very rarely pass context - only if there is really challenging business logic. 

This part should not exist
```
self.context.get('request') and self.context['request'].method in ['POST', 'PUT', 'PATCH']
```